### PR TITLE
Auto complete component

### PR DESCRIPTION
**Pull request by Fatmasiam**
#20  
#### How to test the autoComplete component :
just add import the component in Any page
```javascript 
   <AutoCompleteTags
                  type="skill"
                  palceholder="Choose skills ..."
                  data={[
                    { id: 1, name: 'React' },
                    { id: 2, name: 'Node Js' },
                    { id: 3, name: 'Express JS' },
                  ]}
                />
            
```

### CODE REVIEW

File: client/src/Components/CommonComponents/AutoCompleteTags/index.jsx [hunk 1]
```diff
@@ -1,6 +1,116 @@
-import React from 'react';
-import UnderConstruction from '../../UnderConstruction';
+import React, { Component } from 'react';
+import { Spinner } from 'react-bootstrap';
+import { Typeahead } from 'react-bootstrap-typeahead';
+import PropTypes from 'prop-types';
+import 'react-bootstrap-typeahead/css/Typeahead.css';
+import { skills, offerType } from './staticData';
```
Comment (ashatat) [inline client/src/Components/CommonComponents/AutoCompleteTags/index.jsx]: maybe you should separate importing modules from importing local components and style

File: client/src/Components/CommonComponents/AutoCompleteTags/index.jsx [hunk 2]
```diff
@@ -1,6 +1,116 @@
...
+    const { data, type } = this.props;
+    this.setState({ selectedTags: data });
+
+    if (type === 'skill') {
+      this.setState({ options: skills });
+    }
+    if (type === 'offer_type') {
+      this.setState({ options: offerType });
+    }
+  }
```
Comment (ashatat) [inline client/src/Components/CommonComponents/AutoCompleteTags/index.jsx]: avoid unnecessary rendering, use this.setState once, something like:
```jsx
this.setState(() => {
  let options;
  if (type === 'skill') {
    options === skills;
  } else if (type === 'offer_type') {
    options === offerType;
  }
  return { selectedTags: data, options }
});
```

File: client/src/Components/CommonComponents/AutoCompleteTags/index.jsx [hunk 3]
```diff
@@ -1,6 +1,116 @@
...
+      return this.setState({ allownew: true });
+    });
+  };
+
+  render() {
+    const { options, selectedTags, allownew } = this.state;
+    const { placeholder, type } = this.props;
+    return (
+      <>
+        <h1 className="autocomplete-title">
```
Comment (ashatat) [inline client/src/Components/CommonComponents/AutoCompleteTags/index.jsx]:
```suggestion
        <h1 className="autocomplete__title">
```
BEM methodology, you should also modify it in the style.css

File: client/src/Components/CommonComponents/AutoCompleteTags/index.jsx [hunk 4]
```diff
@@ -1,6 +1,123 @@
...
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Spinner } from 'react-bootstrap';
+import { Typeahead } from 'react-bootstrap-typeahead';
 
-export default function AutoCompleteTags() {
-  return <UnderConstruction />;
+import { skills, offerType } from './staticData';
+
+import 'react-bootstrap-typeahead/css/Typeahead.css';
```
Comment (AhmedAl-Almi) [inline client/src/Components/CommonComponents/AutoCompleteTags/index.jsx]: unnecessary line, if you check the modules you will not find this file.

File: client/src/Components/CommonComponents/AutoCompleteTags/style.css [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  font-weight: normal;
+  opacity: 1;
+  outline: none;
+  padding: 3px 7px;
+  position: absolute;
+  right: 7px;
+  text-shadow: none;
+  top: 2px;
+  margin: 2px;
+}
```
Comment (AhmedAl-Almi) [inline client/src/Components/CommonComponents/AutoCompleteTags/style.css]: add new line

**Pull request merged**